Terraform Crash when creating DigitalOcean droplets

Unfortunately, this happened during a Travis CI build, and so we don't have a ton of access to the logs. They are [here](https://travis-ci.org/CiscoCloud/mantl/jobs/112937843), under the `terraform destroy` step. The Travis configuration is [here](https://github.com/CiscoCloud/mantl/blob/master/.travis.yml), and the terraform file is [here](https://github.com/CiscoCloud/mantl/blob/master/testing/do.tf).

I was able to reproduce this crash by attempting to `terraform` while not connected to the internet. I think it's related. This is on `Terraform v0.6.12`. 
```
module.do-keypair.digitalocean_ssh_key.default: Refreshing state... (ID: 1725781)
Error refreshing state: 1 error(s) occurred:

* digitalocean_ssh_key.default: unexpected EOF
panic: runtime error: invalid memory address or nil pointer dereference
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: [signal 0xb code=0x1 addr=0x0 pc=0x46d5fe]
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: goroutine 37 [running]:
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: panic(0xa6a220, 0xc82000a110)
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 	/opt/go/src/runtime/panic.go:464 +0x3e6
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: github.com/hashicorp/terraform/builtin/providers/digitalocean.resourceDigitalOceanSSHKeyRead(0xc8201ed0e0, 0xaa2be0, 0xc82027ad80, 0x0, 0x0)
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/digitalocean/resource_digitalocean_ssh_key.go:77 +0x24e
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc82027ebc0, 0xc8201dbcb0, 0xaa2be0, 0xc82027ad80, 0xc8202823e8, 0x0, 0x0)
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:209 +0x43a
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc820225110, 0xc820191a80, 0xc8201dbcb0, 0x1, 0x0, 0x0)
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:187 +0x1da
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Refresh(0xc820251340, 0xc82018cba0, 0xc82018ce30, 0x0, 0x0)
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/resource_provider.go:345 +0x6a
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: reflect.Value.call(0xa0eb80, 0xaf5298, 0x13, 0xb42050, 0x4, 0xc8201a7ee8, 0x3, 0x3, 0x0, 0x0, ...)
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 	/opt/go/src/reflect/value.go:435 +0x120d
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: reflect.Value.Call(0xa0eb80, 0xaf5298, 0x13, 0xc8201a7ee8, 0x3, 0x3, 0x0, 0x0, 0x0)
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 	/opt/go/src/reflect/value.go:303 +0xb1
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: net/rpc.(*service).call(0xc82027ec40, 0xc82027ec00, 0xc82027d0e0, 0xc82026e380, 0xc82028e900, 0x923b20, 0xc82018cba0, 0x16, 0x923b80, 0xc82018ce30, ...)
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 	/opt/go/src/net/rpc/server.go:383 +0x1c2
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: created by net/rpc.(*Server).ServeCodec
2016/03/02 15:09:00 [DEBUG] terraform-provider-digitalocean: 	/opt/go/src/net/rpc/server.go:477 +0x49d
2016/03/02 15:09:00 [DEBUG] /usr/bin/terraform-provider-digitalocean: plugin process exited
2016/03/02 15:09:00 [ERROR] root.do-keypair: eval: *terraform.EvalRefresh, err: digitalocean_ssh_key.default: unexpected EOF
2016/03/02 15:09:00 [ERROR] root.do-keypair: eval: *terraform.EvalSequence, err: digitalocean_ssh_key.default: unexpected EOF
2016/03/02 15:09:00 [ERROR] root.do-keypair: eval: *terraform.EvalOpFilter, err: digitalocean_ssh_key.default: unexpected EOF
2016/03/02 15:09:00 [ERROR] root.do-keypair: eval: *terraform.EvalSequence, err: digitalocean_ssh_key.default: unexpected EOF
2016/03/02 15:09:00 [TRACE] [walkRefresh] Exiting eval tree: digitalocean_ssh_key.default
2016/03/02 15:09:00 [DEBUG] vertex module.do-keypair.output.keypair_id, got dep: module.do-keypair.digitalocean_ssh_key.default
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.control, got dep: module.do-keypair.output.keypair_id
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.control, got dep: module.do-keypair.digitalocean_ssh_key.default
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.control, got dep: module.do-hosts.var.region_name
2016/03/02 15:09:00 [DEBUG] vertex provider.module.do-keypair.digitalocean (close), got dep: module.do-keypair.digitalocean_ssh_key.default
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.edge, got dep: module.do-keypair.output.keypair_id
2016/03/02 15:09:00 [DEBUG] vertex provider.module.do-keypair.digitalocean (close), got dep: module.do-keypair.provider.digitalocean
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.var.ssh_key, got dep: module.do-keypair.output.keypair_id
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.edge, got dep: module.do-hosts.digitalocean_droplet.edge (destroy)
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.var.ssh_key, got dep: module.do-keypair.digitalocean_ssh_key.default
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.edge, got dep: module.do-hosts.var.region_name
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-keypair.output.keypair_id
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.edge, got dep: module.do-hosts.var.edge_size
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.control, got dep: module.do-hosts.var.ssh_key
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-keypair.digitalocean_ssh_key.default
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.edge, got dep: module.do-hosts.var.image_name
2016/03/02 15:09:00 [DEBUG] vertex provider.module.do-hosts.digitalocean (close), got dep: module.do-hosts.digitalocean_droplet.control
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-hosts.digitalocean_droplet.worker (destroy)
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-hosts.provider.digitalocean
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.output.control_ips, got dep: module.do-hosts.digitalocean_droplet.control
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-hosts.var.worker_size
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-hosts.var.short_name
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.do-hosts.output.control_ips
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-hosts.var.image_name
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-hosts.var.ssh_key
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-hosts.var.datacenter
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-hosts.var.region_name
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.worker, got dep: module.do-hosts.var.worker_count
2016/03/02 15:09:00 [DEBUG] vertex provider.module.do-hosts.digitalocean (close), got dep: module.do-hosts.digitalocean_droplet.worker
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.edge, got dep: module.do-keypair.digitalocean_ssh_key.default
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.edge, got dep: module.do-hosts.var.short_name
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.digitalocean_droplet.edge, got dep: module.do-hosts.var.ssh_key
2016/03/02 15:09:00 [DEBUG] vertex provider.module.do-hosts.digitalocean (close), got dep: module.do-hosts.digitalocean_droplet.edge
2016/03/02 15:09:00 [DEBUG] vertex provider.module.do-hosts.digitalocean (close), got dep: module.do-hosts.digitalocean_droplet.control (destroy)
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.output.edge_ips, got dep: module.do-hosts.digitalocean_droplet.edge
2016/03/02 15:09:00 [DEBUG] vertex provider.module.do-hosts.digitalocean (close), got dep: module.do-hosts.digitalocean_droplet.worker (destroy)
2016/03/02 15:09:00 [DEBUG] vertex module.do-hosts.output.worker_ips, got dep: module.do-hosts.digitalocean_droplet.worker
2016/03/02 15:09:00 [DEBUG] vertex provider.module.do-hosts.digitalocean (close), got dep: module.do-hosts.digitalocean_droplet.edge (destroy)
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.do-hosts.output.worker_ips
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.do-hosts.digitalocean_droplet.worker
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: provider.module.do-hosts.digitalocean (close)
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.aws-elb.output.zone_id (orphan)
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.do-hosts.plan-destroy
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.traefik-elb.output.zone_id (orphan)
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.traefik-elb.output.fqdn (orphan)
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.do-hosts.digitalocean_droplet.edge
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: provider.module.do-keypair.digitalocean (close)
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.aws-elb.output.fqdn (orphan)
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.do-hosts.output.edge_ips
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.do-hosts.digitalocean_droplet.control
2016/03/02 15:09:00 [DEBUG] vertex root, got dep: module.do-keypair.plan-destroy
2016/03/02 15:09:00 [DEBUG] waiting for all plugin processes to complete...



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -20,6 +20,7 @@ BUG FIXES:
   * core: Modules sourced from a Mercurial repository now work correctly on Windows [GH-5739]
   * provider/aws: Fix crash when an `aws_rds_cluster_instance` is removed outside of Terraform [GH-5717]
   * provider/aws: `aws_lambda_function` resources no longer error on refresh if deleted externally to Terraform [GH-5668]
+  * provider/aws: `aws_vpn_connection` resources deleted via the console on longer cause a crash [GH-5747]
   * provider/google: Default description `google_dns_managed_zone` resources to "Managed By Terraform" [GH-5428]
   * provider/google: Fix error message on invalid instance URL for `google_compute_instance_group` [GH-5715]
 

--- a/builtin/providers/digitalocean/resource_digitalocean_ssh_key.go
+++ b/builtin/providers/digitalocean/resource_digitalocean_ssh_key.go
@@ -74,7 +74,7 @@ func resourceDigitalOceanSSHKeyRead(d *schema.ResourceData, meta interface{}) er
 	if err != nil {
 		// If the key is somehow already destroyed, mark as
 		// successfully gone
-		if resp.StatusCode == 404 {
+		if resp != nil && resp.StatusCode == 404 {
 			d.SetId("")
 			return nil
 		}

